app/jobs/unzip_job.rb

```diff
@@ -8,17 +8,25 @@ class UnzipJob < BaseDepositJob
 
   def perform(work_version)
     zip_attached_file = work_version.attached_files.last
-    destroy_attached_files_except(zip_attached_file, work_version)
-    Zip::File.open(filepath_for(zip_attached_file)) do |zip_file|
-      Dir.mktmpdir do |temp_dir|
-        zip_file.each do |entry|
-          next if ignore?(entry)
-
-          attach_file(entry:, temp_dir:, work_version:)
+    # Ensure the last file uploaded is actually a ZIP file.
+    # This guards against a user selecting the zip option after having previously uploaded non-zip files
+    # but then not actually uploading a zip file.  Previously this would have resulted in
+    # this job getting stuck.  See https://github.com/sul-dlss/happy-heron/issues/3454
+    # Instead, we want to just revert back to the browser upload style
+    # without doing anything else.
+    if zip_attached_file.content_type == 'application/zip'
```

Comment:
basically check for this before doing the actually unzipping and destroying of the rest of the files